analysis/src/code.rs

```diff
@@ -0,0 +1,152 @@
+//! This module abstracts various analysis of source code for a given package
+
+use anyhow::{anyhow, Result};
+use camino::Utf8Path;
+use guppy::graph::{DependencyDirection, PackageGraph, PackageMetadata};
+use serde::{Deserialize, Serialize};
+use std::{cell::RefCell, collections::HashMap, iter};
+use tokei::{Config, LanguageType, Languages};
+
+#[derive(Debug, Clone)]
+pub struct CodeReport {
+    pub name: String,
+    pub version: String,
+    pub is_direct: bool,
+    pub loc_report: LOCReport,
+    pub dep_report: DepReport,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct LOCReport {
+    pub is_analyzed: bool,
+    pub total_code_loc: u64, // excludes comment and white lines
+    pub rust_code_loc: u64,  // excludes comment and white lines
+}
+
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct DepReport {
+    pub is_analyzed: bool,
+    pub total_deps: u64,
+    pub deps_total_code_loc: u64,
```

Comment:
Just a nit but this reads strange to me because the `c` in `loc` is code :) Maybe just `deps_total_loc`?